service/src/main/kotlin/io/provenance/api/domain/usecase/provenance/tx/scope/ChangeScopeOwnership.kt

```diff
@@ -0,0 +1,81 @@
+import io.provenance.api.domain.usecase.AbstractUseCase
+import io.provenance.api.domain.usecase.common.errors.NotFoundError
+import io.provenance.api.domain.usecase.common.originator.EntityManager
+import io.provenance.api.domain.usecase.provenance.account.GetSigner
+import io.provenance.api.domain.usecase.provenance.account.models.GetSignerRequest
+import io.provenance.api.domain.usecase.provenance.tx.scope.models.ChangeScopeOwnershipRequestWrapper
+import io.provenance.api.frameworks.config.ProvenanceProperties
+import io.provenance.api.frameworks.provenance.extensions.toAny
+import io.provenance.api.frameworks.provenance.extensions.toBase64String
+import io.provenance.api.frameworks.provenance.extensions.toMessageSet
+import io.provenance.api.frameworks.provenance.extensions.toTxBody
+import io.provenance.api.models.p8e.TxBody
+import io.provenance.api.models.p8e.TxResponse
+import io.provenance.client.protobuf.extensions.isSet
+import io.provenance.metadata.v1.MsgWriteScopeRequest
+import io.provenance.scope.util.ProtoJsonUtil.toJson
+import org.springframework.stereotype.Component
+
+@Component
+class ChangeScopeOwnership(
+    private val provenance: Provenance,
+    private val entityManager: EntityManager,
+    private val getSigner: GetSigner,
+    private val provenanceProperties: ProvenanceProperties,
+) : AbstractUseCase<ChangeScopeOwnershipRequestWrapper, TxResponse>() {
+    override suspend fun execute(args: ChangeScopeOwnershipRequestWrapper): TxResponse {
+        require(args.request.newValueOwner !== null || args.request.newDataAccess !== null) {
+            "Must request at least one change to the scope"
+        }
+
+        val signer = getSigner.execute(GetSignerRequest(args.uuid, args.request.account))
+
+        val scopeResponse = provenance.getScope(args.request.provenanceConfig, args.request.scopeId)
+            .takeIf { response -> response.scope.isSet() }
+            ?: throw NotFoundError("No scope found")
+
+        scopeResponse.scope.scopeIdInfo.scopeUuid.let { actualScopeId ->
+            require(actualScopeId === args.request.scopeId.toString()) {
+                "Expected to fetch a scope with UUID ${args.request.scopeId} but got a UUID of $actualScopeId"
+            }
+        }
+
+        val message = MsgWriteScopeRequest.newBuilder().apply {
+            scopeUuid = args.request.scopeId.toString()
+            specUuid = scopeResponse.scope.scopeSpecIdInfo.scopeSpecUuid
+            scope = scopeResponse.scope.scope.toBuilder().also { scopeBuilder ->
+                args.request.newValueOwner?.let { requestedNewValueOwner ->
+                    /** Only change value owner if [valueOwnerAddress][args.request.valueOwnerAddress] was not null. */
+                    scopeBuilder.valueOwnerAddress = requestedNewValueOwner
+                }
+                /** Only change data access if [newDataAccess][args.request.newDataAccess] was not null. */
```

Comment:
I suppose we can remove the data access change from this usecase, but my thought was that it might be convenient to allow data access changes within the same endpoint. Someone wishing to do so would have to before they give up the value ownership, right?